test/SlowTests/Client/Subscriptions/CriteriaScript.cs

```diff
@@ -128,32 +132,27 @@ select project(d)
                     };
 
                     var subsId = subscriptionManager.Create(subscriptionCreationParams);
-                    using (var subscription = subscriptionManager.GetSubscriptionWorker<BlittableJsonReaderObject>(new SubscriptionWorkerOptions(subsId)))
+                    using (var subscription = subscriptionManager.GetSubscriptionWorker<Thing>(new SubscriptionWorkerOptions(subsId)))
```

Comment:
This looks like something that user would do?

Reply:
Yeah, that's an interesting question, because "formally", we say that in blittable and contextes we do not support parallel usage, but "intuitively" it seems "ok" to be able to read blittables parallelly, the problem is that when we access strings in a blittable, we use the context's LazyStringValue object pool, and therefore it's not possible to do that, unless we disconnect blittable from the context..

Reply:
First, we dont expect the average user to use blittables. Second, there will always be pitfalls like that, for example, if we had another context.read outside the subscription task, in that case there is no way we can accept that. We could add flag to blittable reader and to context.read that will tell them not to use object pooling and other non mandatory context services, but we have to remember that it won't prevent all pitfalls and we should probably document common ones..